File: app/src/main/java/com/robotnec/budget/core/mvp/presenter/AccountsPresenter.java
package com.robotnec.budget.core.mvp.presenter;

import javax.inject.Inject;

import com.robotnec.budget.core.dao.AccountDao;
import com.robotnec.budget.core.di.ApplicationComponent;
import com.robotnec.budget.core.mvp.view.AccountsView;
import com.robotnec.budget.core.navigator.NavigationBundle;
import com.robotnec.budget.core.navigator.Navigator;

/**
 * @author zak <zak@swingpulse.com>
 */
public class AccountsPresenter extends Presenter<AccountsView> {

    @Inject
    AccountDao accountDao;

    @Inject
    Navigator navigator;

    public AccountsPresenter(AccountsView view) {
        super(view);
    }

    @Override
    public void injectComponent(ApplicationComponent applicationComponent) {
        applicationComponent.inject(this);
    }

    @Override
    public void onViewReady() {
        view.displayAccounts(accountDao.getAll());
    }

    public void addOrUpdateAccount(NavigationBundle navigationBundle) {
        navigator.openAddAccountScreen(navigationBundle);
    }
}

File: app/src/main/java/com/robotnec/budget/core/mvp/presenter/AccountsPresenter.kt
package com.robotnec.budget.core.mvp.presenter

import com.robotnec.budget.refactored.di.ApplicationComponent
import com.robotnec.budget.core.mvp.view.AccountsView
import com.robotnec.budget.core.navigator.NavigationBundle
import com.robotnec.budget.core.navigator.Navigator
import com.robotnec.budget.core.persistence.dao.AccountDao
import javax.inject.Inject

/**
 * @author zak zak@swingpulse.com>
 */
class AccountsPresenter(view: AccountsView) : Presenter<AccountsView>(view) {

    @Inject
    internal lateinit var accountDao: AccountDao

    @Inject
    internal lateinit var navigator: Navigator

    override fun injectComponent(applicationComponent: ApplicationComponent) {
        applicationComponent.inject(this)
    }

    override fun onViewResume() {
        view.displayAccounts(accountDao.all)
    }

    fun addOrUpdateAccount(navigationBundle: NavigationBundle<*>) {
        navigator.openAddAccountScreen(navigationBundle)
    }
}
